flocker/node/agents/functional/test_ebs.py

```diff
@@ -182,6 +188,128 @@ def test_next_device_in_use(self):
+                          MandatoryProfiles.GOLD,
+                          size_GiB=1024*1024)
+
+    def test_create_volume_silver_profile(self):
+        """
+        Requesting ``silver`` profile during volume creation honors
+        ``silver`` attributes.
+        """
+        self._assert_create_volume_with_mandatory_profile(
+            MandatoryProfiles.SILVER)
+
+    def test_create_too_large_volume_silver_profile(self):
+        """
+        Too large volume (> 16TiB) for ``silver`` profile.
+        """
+        self.assertRaises(EC2ResponseError,
+                          self._assert_create_volume_with_mandatory_profile,
+                          MandatoryProfiles.SILVER,
+                          size_GiB=1024*1024)
+
+    def test_create_largest_volume_silver_profile(self):
+        """
+        Largest volume with ``silver`` profile.
+        """
+        raise SkipTest(u'This test is expensive (money).')
+        self._assert_create_volume_with_mandatory_profile(
+            MandatoryProfiles.SILVER, size_GiB=16*1024)
+
+    def test_create_volume_bronze_profile(self):
+        """
+        Requesting ``bronze`` profile during volume creation honors
+        ``bronze`` attributes.
+        """
+        self._assert_create_volume_with_mandatory_profile(
+            MandatoryProfiles.BRONZE)
+
+    def test_create_largest_volume_bronze_profile(self):
+        """
+        Largest volume with ``bronze`` profile.
+        """
+        raise SkipTest(u'This test is expensive (money).')
+        self._assert_create_volume_with_mandatory_profile(
+            MandatoryProfiles.BRONZE, size_GiB=1024)
+
+    def _assert_create_volume_with_mandatory_profile(self, profile,
+                                                     created_profile=None,
+                                                     size_GiB=4):
+        """
+        Volume created with given profile has the attributes
+        expected from the profile.
+
+        :param ValueConstant profile: Name of profile to use for creation.
+        :param ValueConstant created_profile: Name of the profile volume is
+            expected to be created with.
+        :param int size_GiB: Size of volume to be created.
+        """
+        if created_profile is None:
+            created_profile = profile
+        volume1 = self.api.create_volume_with_profile(
+            dataset_id=uuid4(),
+            size=self.minimum_allocatable_size * size_GiB,
+            profile_name=profile.value)
+
+        cannonical_profile = MandatoryProfiles.lookupByValue(
+            created_profile.value)
+        A = EBSMandatoryProfileAttributes.lookupByName(
+            cannonical_profile.name).value
+        ebs_volume = self.api._get_ebs_volume(volume1.blockdevice_id)
+        self.assertEqual(ebs_volume.type, A.volume_type.value)
+        requested_iops = A.requested_iops(ebs_volume.size)
+        if requested_iops is not None:
```

Comment:
Thanks for pointing out!

i like this better:
```
self.assertEqual(ebs_volume.iops if requested_iops is not None else None, requested_iops)
```
because requested_iops are not meant to specified for certain poor profiles. Changed. Thanks!